virttest/test_setup.py

```diff
@@ -1911,6 +1911,20 @@ def __init__(self, params):
             else:
                 self.attr = []
 
+            # The 'connect_driver' could be set to a different value for
+            # different libvirt daemon modes.
+            # For example, it should be "QEMU" for the monolithic libvirtd
+            # environment and "network" for the modular daemons environment,
+            # it should set to "connect_driver:QEMU|network".
+            for idx in range(len(self.attr)):
```

Comment:
one way to go forward may be like this:
after the line 1910, this may help avoid code changes in tp-libvirt
```

action_lookup_str = self.params.get("action_lookup")
if 'network' in action_lookup_str:
   action_lookup_str = action_lookup_str.replace('qemu, '' network")
elif  ''nwfilter' in action_lookup_str:
   action_lookup_str = action_lookup_str.replace('qemu, '' nwfilter")
self.attr = action_lookup_str.split()
```

Reply:
Thanks for your advice! But I prefer my current solution. First of all, trying to get the driver from "action_lookup" string is not rigorous. Second, even though I list all the drivers by 'if/elif' here, I still need to change tp-libvirt to make the cases pass.